src/test/java/io/phasetwo/keycloak/events/WebhookSenderEventListenerProviderTest.java

```diff
@@ -0,0 +1,281 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.xgp.http.server.Server;
+import com.google.common.collect.ImmutableSet;
+
+import io.phasetwo.keycloak.representation.ExtendedAdminEvent;
+import io.phasetwo.keycloak.resources.AbstractResourceTest;
+import jakarta.ws.rs.core.Response;
+import lombok.extern.jbosslog.JBossLog;
+
+@JBossLog
+public class WebhookSenderEventListenerProviderTest extends AbstractResourceTest {
+  final static String TEST_REALM = "testRealm";
+  CloseableHttpClient httpClient = HttpClients.createDefault();
+
+  String webhookUrl(String realm) {
+    return getAuthUrl() + "/realms/" + realm + "/webhooks";
+  }
+
+  String usersUrl(String realm) {
+    return getAuthUrl() + "/admin/realms/" + realm + "/users";
+  }
+
+  @Test
+  public void testAdminEventContainsCorrectRealmMasterMaster() throws Exception {
+    // Configure
+    RealmResource realm = keycloak.realm(REALM);
+    RealmRepresentation realmRepresentation = realm.toRepresentation();
+    realmRepresentation.setAdminEventsEnabled(true);
+    realmRepresentation.setAdminEventsDetailsEnabled(true);
+    realmRepresentation.setEventsListeners(Arrays.asList("ext-event-webhook"));
+    realm.update(realmRepresentation);
+
+    AtomicReference<String> body = new AtomicReference<String>();
+    // create a server on a free port with a handler to listen for the event
+    int port = WEBHOOK_SERVER_PORT;
+    String webhookId = createWebhook(
+        keycloak,
+        httpClient,
+        webhookUrl(REALM),
+        "http://host.testcontainers.internal:" + port + "/webhook",
+        "qlfwemke",
+        ImmutableSet.of("admin.*"));
+
+    Server server = new Server(port);
+    server
+        .router()
+        .POST(
+            "/webhook",
+            (request, response) -> {
+              String r = request.body();
+              log.infof("%s", r);
+              body.set(r);
+              response.body("OK");
+              response.status(202);
+            });
+    server.start();
+    Thread.sleep(1000l);
+
+    try {
+	    // cause an event to be sent
+	    UserRepresentation userRepresentation = new UserRepresentation();
+	    userRepresentation.setUsername("username");
+	    Response userResponse = realm.users().create(userRepresentation);
+	    assertThat(userResponse.getStatus(), is(201));
+
+	    Thread.sleep(1000l);
+
+	    // check the handler for the event, after a delay
+	    String receivedPayload = body.get();
+	    ExtendedAdminEvent event = parseEvent(receivedPayload);
+	    assertThat(event.getRealmId(), equalTo(REALM));
+	    assertThat(event.getAuthDetails().getRealmId(), equalTo(REALM));
+    }
+    finally {
+        server.stop();
+        removeWebhook(keycloak,
+                httpClient,
+                webhookUrl(REALM),
+                webhookId);
+    }
+  }
+
+  @Test
+  public void testAdminEventContainsCorrectRealmMasterTest() throws Exception {
```

Comment:
Could we add another test in which the user which the user that generates the event is part of TEST_REALM?

I believe you can create a new user in the TEST_REALM and login by using:
`        
  CredentialRepresentation pass = new CredentialRepresentation();
    pass.setType(CredentialRepresentation.PASSWORD);
    pass.setValue(password);
    pass.setTemporary(false);
    UserRepresentation user = new UserRepresentation();
    user.setEnabled(true);
    user.setRealmRoles(List.of("role1", "role2"));
    user.setUsername(username);
    if (email != null) {
      user.setEmail(email);
    }
    user.setCredentials(ImmutableList.of(pass));
    keycloak.realm(realm).users().create(user);
    return keycloak.realm(realm).users().search(user.getUsername()).get(0);
    var keycloak2 = Keycloak.getInstance(container.getAuthServerUrl(), "admin-cli", "user1", "pass");
`

For more info check the approach in this test: https://github.com/p2-inc/keycloak-orgs/blob/main/src/test/java/io/phasetwo/service/resource/OrganizationResourceTest.java#L1622